cmake/configs/modes.cmake

```diff
@@ -9,104 +9,173 @@
+        "Flags used by the C compiler for ${mode} build type or configuration." FORCE)
+
+    set(CMAKE_CXX_FLAGS_${build_mode}
+        "${DEFINE_BUILD_CXX_COMPILER_FLAGS}" CACHE STRING
+        "Flags used by the C++ compiler for ${mode} build type or configuration." FORCE)
+
+    set(CMAKE_EXE_LINKER_FLAGS_${build_mode}
+        "${DEFINE_BUILD_LINK_FLAGS}" CACHE STRING
+        "Linker flags to be used to create executables for ${mode} build type." FORCE)
+
+    set(CMAKE_SHARED_LINKER_FLAGS_${build_mode}
+        "${DEFINE_BUILD_LINK_FLAGS}" CACHE STRING
+        "Linker lags to be used to create shared libraries for ${mode} build type." FORCE)
 
     mark_as_advanced(
-        CMAKE_CXX_FLAGS_ASAN
-        CMAKE_C_FLAGS_ASAN
-        CMAKE_EXE_LINKER_FLAGS_ASAN
-        CMAKE_SHARED_LINKER_FLAGS_ASAN
+        CMAKE_CXX_FLAGS_${build_mode}
+        CMAKE_C_FLAGS_${build_mode}
+        CMAKE_EXE_LINKER_FLAGS_${build_mode}
+        CMAKE_SHARED_LINKER_FLAGS_${build_mode}
     )
-    list(APPEND build_modes "ASan")
-endif()
+    set(BUILD_MODES "${BUILD_MODES};${mode}" CACHE INTERNAL "")
+endfunction()
 
-# Create an UBSAN build variant
-if("${CMAKE_C_COMPILER_ID}" STREQUAL "MSVC")
-    set(ubsan_link_flags "/fsanitize=undefined")
-    set(ubsan_compiler_flag "/fsanitize=undefined")
-elseif("${CMAKE_C_COMPILER_ID}" STREQUAL "Clang" OR "${CMAKE_C_COMPILER_ID}" STREQUAL "AppleClang")
-    set(ubsan_link_flags "-fsanitize=undefined")
-    set(ubsan_compiler_flag "-fsanitize=undefined")
+if(MSVC)
+    set(no_omit_frame_flag "/Oy-")
 else()
-    set(ubsan_link_flags "-fsanitize=undefined -lubsan")
-    set(ubsan_compiler_flag "-fsanitize=undefined")
+    set(no_omit_frame_flag "-fno-omit-frame-pointer")
+    set(no_omit_frame_flag "-fno-omit-frame-pointer")
 endif()
 
-# Needs to validate linker flags for the test to also pass.
-set(CMAKE_REQUIRED_FLAGS "${ubsan_link_flags}")
-# Check if the UBSAN compiler flag is available.
-check_c_compiler_flag("${ubsan_compiler_flag}" HAVE_UB_SANITIZER)
-unset(CMAKE_REQUIRED_FLAGS)
-
-if(HAVE_UB_SANITIZER)
-    set(CMAKE_C_FLAGS_UBSAN
-        "${CMAKE_C_FLAGS_DEBUG} ${ubsan_compiler_flag} ${no_omit_frame_flag}" CACHE STRING
-        "Flags used by the C compiler for UBSan build type or configuration." FORCE)
-
-    set(CMAKE_CXX_FLAGS_UBSAN
-        "${CMAKE_CXX_FLAGS_DEBUG} ${ubsan_compiler_flag} ${no_omit_frame_flag}" CACHE STRING
-        "Flags used by the C++ compiler for UBSan build type or configuration." FORCE)
-
-    set(CMAKE_EXE_LINKER_FLAGS_UBSAN
-        "${CMAKE_SHARED_LINKER_FLAGS_DEBUG} ${ubsan_link_flags}" CACHE STRING
-        "Linker flags to be used to create executables for UBSan build type." FORCE)
+# ASAN build variant flags.
+if(MSVC)
+    set(asan_link_c_flags "/fsanitize=address")
+    set(asan_compiler_c_flag "/fsanitize=address")
+    set(asan_compiler_cxx_flag "/fsanitize=address")
+else()
+    set(asan_link_flags "-fsanitize=address")
```

Comment:
The variable name here is different from that of line 115 above. I guess this one is correct?

Reply:
Good catch! Line 115 should be `asan_link_flags`. Fixed in ba5e02d